Tangem/Common/OrganizeWalletModels/OrganizeTokensOptionsManager.swift

```diff
@@ -0,0 +1,105 @@
+import CombineExt
+
+final class OrganizeTokensOptionsManager {
+    private let userTokenListManager: UserTokenListManager
+    private let editingThrottleInterval: TimeInterval
+    private let editedGroupingOption = PassthroughSubject<OrganizeTokensOptions.Grouping, Never>()
+    private let editedSortingOption = PassthroughSubject<OrganizeTokensOptions.Sorting, Never>()
+
+    init(
+        userTokenListManager: UserTokenListManager,
+        editingThrottleInterval: TimeInterval
+    ) {
+        self.userTokenListManager = userTokenListManager
+        self.editingThrottleInterval = editingThrottleInterval
+    }
+}
+
+// MARK: - OrganizeTokensOptionsProviding protocol conformance
+
+extension OrganizeTokensOptionsManager: OrganizeTokensOptionsProviding {
+    var groupingOption: AnyPublisher<OrganizeTokensOptions.Grouping, Never> {
+        let editedGroupingOption = editedGroupingOption
+            .throttle(
+                for: RunLoop.SchedulerTimeType.Stride(editingThrottleInterval),
+                scheduler: RunLoop.main,
+                latest: false
+            )
+            .eraseToAnyPublisher()
+
+        let groupingOptionFromUserTokenList = userTokenListManager
+            .userTokenList
+            .prefix(untilOutputFrom: editedGroupingOption)
+            .map { OrganizeTokensOptionsConverter().convert($0.group) }
+            .eraseToAnyPublisher()
+
+        return [
+            editedGroupingOption,
+            groupingOptionFromUserTokenList,
+        ].merge()
+    }
+
+    var sortingOption: AnyPublisher<OrganizeTokensOptions.Sorting, Never> {
+        let editedSortingOption = editedSortingOption
+            .throttle(
+                for: RunLoop.SchedulerTimeType.Stride(editingThrottleInterval),
+                scheduler: RunLoop.main,
+                latest: false
+            )
+            .eraseToAnyPublisher()
+
+        let sortingOptionFromUserTokenList = userTokenListManager
+            .userTokenList
+            .prefix(untilOutputFrom: editedSortingOption)
+            .map { OrganizeTokensOptionsConverter().convert($0.sort) }
+            .eraseToAnyPublisher()
+
+        return [
+            editedSortingOption,
+            sortingOptionFromUserTokenList,
+        ].merge()
+    }
+}
+
+// MARK: - OrganizeTokensOptionsEditing protocol conformance
+
+extension OrganizeTokensOptionsManager: OrganizeTokensOptionsEditing {
+    func group(by groupingOption: OrganizeTokensOptions.Grouping) {
+        editedGroupingOption.send(groupingOption)
+    }
+
+    func sort(by sortingOption: OrganizeTokensOptions.Sorting) {
+        editedSortingOption.send(sortingOption)
+    }
+
+    func save() -> AnyPublisher<Void, Never> {
+        return .just
+            .withLatestFrom(groupingOption, sortingOption, userTokenListManager.userTokenList)
+            .withWeakCaptureOf(self)
+            .flatMapLatest { input in
+                let (manager, (grouping, sorting, userTokenList)) = input
```

Comment:
```suggestion
            .flatMapLatest { [weak self] grouping, sorting, userTokenList in
```

написано вроде прикольно, но как будто привычней такой вариант )

Reply:
Мне кажется `withWeakCaptureOf` весьма удобно:
1. Не надо делать лишний анврап
2. Операторы ниже по стриму просто не вызовятся, если слабо захваченный объект уже умрет на этот момент - вместо выполнения бесполезной работы